.github/workflows/tests.yaml

```diff
@@ -78,13 +90,24 @@ jobs:
         run: |
           pip install -U pip wheel
           pip install .[dev]
-      - name: Test with pytest
+
+      - name: Test with pytest (with coverage)
+        if: matrix.run-coverage == true
+        timeout-minutes: 25
+        run: |
+          pytest -v -n 2 --headless --browser=${{ matrix.browser }} --pf-version=${{ matrix.pf-version }} --cov=./ --cov-report=xml --reruns 2 --reruns-delay 5
```

Comment:
Consider (separate PR) adding a hatch config so we can just use `hatch run test` and `hatch run test-cov` and can set many of these options as default.